src/app/core/services/videos-list.service.ts

```diff
@@ -0,0 +1,54 @@
+import { Injectable } from '@angular/core';
+import { Observable, BehaviorSubject } from 'rxjs';
+
+import { Video } from '@core/models/video.model';
+import { LocalStorageService } from './local-store.service';
+import { NewVideoService } from './new-video.service';
+import demo from '@tools/demo';
+
+@Injectable({
+	providedIn: 'root',
+})
+export class VideosListService {
+	private videos: Video[] = JSON.parse(localStorage.getItem('videos') || '[]') as Video[];
+
+	private videosBehaviorSubject = new BehaviorSubject<Video[]>(this.videos);
+	public videos$: Observable<Video[]> = this.videosBehaviorSubject.asObservable();
+
+	constructor(public newVideoService: NewVideoService, public localStorageService: LocalStorageService) {}
+
+	public adVideo(video: Video): void {
+		this.videos.push(video);
+		this.setVidoes();
+	}
+
+	public deleteVideo(index: number): void {
+		this.videos.splice(index, 1);
+		this.setVidoes();
+	}
+
+	public toogleFavorite(id: number | string): void {
+		this.videos = this.videos.map(item => {
```

Comment:
mapa jest wygodna ale to jedna z cięższych metod, a tutaj mapujemy przez całą tablicę żeby zaznaczyć tylko jeden element, może znajdziemy go po indexie ? Nie chodzi mi o to że wykorzystanie map() jest złe ale chcę zwrócić tylko uwagę że performance też jest ważną kwestią https://leanylabs.com/blog/js-forEach-map-reduce-vs-for-for_of/